File: java/lib/src/main/java/com/svix/models/DestinationStatus.java
// This file is @generated
package com.svix.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.Utils.ToQueryParam;

public enum DestinationStatus implements ToQueryParam {
    ENABLED("enabled"),
    PAUSED("paused"),
    DISABLED("disabled"),
    RETRYING("retrying");
    private final String value;

    DestinationStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: javascript/src/models/destinationStatus.ts
// this file is @generated

export enum DestinationStatus {
  Enabled = "enabled",
  Paused = "paused",
  Disabled = "disabled",
  Retrying = "retrying",
}

export const DestinationStatusSerializer = {
  _fromJsonObject(object: any): DestinationStatus {
    return object;
  },

  _toJsonObject(self: DestinationStatus): any {
    return self;
  },
};
